src/main/java/atrium/AtriumClient.java

```diff
@@ -0,0 +1,526 @@
+      System.exit(0);
+      return "Error";
+    }
+
+    int i = 0;
+    try {
+      String uri = environment + endpoint;
+      URL url = new java.net.URL(uri);
+
+      HttpsURLConnection con = (HttpsURLConnection)url.openConnection();
+      con.setRequestMethod(mode);
+      con.setRequestProperty("Accept", "application/vnd.mx.atrium.v1+json");
+      con.setRequestProperty("Content-Type", "application/json");
+      con.setRequestProperty("MX-API-Key", mxAPIKEY);
+      con.setRequestProperty("MX-Client-ID", mxCLIENTID);
+
+      if (!body.equals("")) {
+        con.setDoOutput(true);
+        OutputStream out = con.getOutputStream();
+        out.write(body.getBytes("UTF-8"));
+        out.close();
+      }
+
+      httpError(con.getResponseCode());
+
+      BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+
+      StringBuilder response = new StringBuilder();
+      String inputLine;
+      while ((inputLine = in.readLine()) != null) {
+        response.append(inputLine);
+      }
+      in.close();
+      con.disconnect();
+
+      return response.toString();
+    }
+    catch (Exception e) {
+      e.printStackTrace();
+    }
+    return Integer.toString(i);
+  }
+
+  // Print and exit on http error
+  private void httpError(int code)
+  {
+    try
+    {
+      switch (code) {
+      case 400:
+        System.out.println(code + " error: Required parameter is missing.");
+        break;
+      case 401:
+        System.out.println(code + " error: Invalid MX-API-Key, MX-Client-ID, or being used in wrong environment.");
+        break;
+      case 403:
+        System.out.println(code + " error: Requests must be HTTPS.");
+        break;
+      case 404:
+        System.out.println(code + " error: GUID / URL path not recognized.");
+        break;
+      case 405:
+        System.out.println(code + " error: Endpoint constraint not met.");
+        break;
+      case 406:
+        System.out.println(code + " error: Specifiy valid API version.");
+        break;
+      case 409:
+        System.out.println(code + " error: Object already exists.");
+        break;
+      case 422:
+        System.out.println(code + " error: Data provided cannot be processed.");
+        break;
+      case 500:
+      case 502:
+      case 504:
+        System.out.println(code + " error: An unexpected error occurred on MX's systems.");
+      case 503:
+        System.out.println(code + " error: Please try again later. The MX Platform is currently being updated.");
+      }
+
+      if (code / 100 == 4 || code / 100 == 5) {
+        System.exit(0);
+      }
+    }
+    catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private String optionalParameters(String name, String fromDate, String toDate, String pageNumber, String recordsPerPage) {
+    String params = "?";
```

Comment:
While I'd prefer this get axed and the params all get checked in each method, there is a much more memory efficient way to do string concatenation. Java's strings are immutable, and every time you concat strings, it creates new copies. With how this is set up, every param that isn't empty will end up creating ~4 new strings in memory. To do this without creating tons of temporary strings in memory, switch to using a `StringBuilder`.